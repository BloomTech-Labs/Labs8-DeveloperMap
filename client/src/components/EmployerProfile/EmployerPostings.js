import React from 'react';
import firebase from '../../firebase/firebase';
import JobPosting from './JobPosting';
import FavHeart from '../../images/favorites-icon.png';
import Heart from '../../images/hollow-heart.png';
import axios from 'axios';

import { Posts, PostContainer } from './EmployerStyles';

class EmployerPostings extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  // favToggle = async (e, post) => {
  //   e.preventDefault();

  //   const user = this.props.user;
  //   if (!user) {
  //     this.props.history.push('/signin');
  //   } else if (e.target.src === FavHeart) {
  //     //Removes Favorited Post from current User if confirms
  //     const favoritedList = [
  //       ...this.state.favoritedList.filter(jobId => jobId !== e.target.id),
  //     ];
  //     this.setState({ favoritedList });
  //     e.target.src = Heart;
  //   } else {
  //     //Adds Favorited Post to current User's list if confirms
  //     const favoritedList = [...this.state.favoritedList];
  //     favoritedList.push(e.target.id);
  //     this.setState({ favoritedList });
  //     e.target.src = FavHeart;
  //   }
  // };

  render() {
    const { posts, favoritedList } = this.props;
    return (
      <PostContainer>
        {posts ? (
          <Posts>
            {Object.values(posts).map((post, i) => (
              <JobPosting
                key={i}
                index={i}
                favoritedList={favoritedList}
                {...this.props}
                post={post}
                favToggle={this.props.favToggle}
              />
            ))}
          </Posts>
        ) : (
          <Posts>
            <h1>This company has no job postings.</h1>
          </Posts>
        )}
      </PostContainer>
    );
  }
}

export default EmployerPostings;
