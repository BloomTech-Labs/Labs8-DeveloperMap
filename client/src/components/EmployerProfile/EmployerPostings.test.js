import React from 'react';
import renderer from 'react-test-renderer';
import EmployerPostings from './EmployerPostings';

jest.mock('../../firebase/firebase', () => {
  const firebasemock = require('firebase-mock');

  const mockdatabase = new firebasemock.MockFirebase();
  const mockauth = new firebasemock.MockFirebase();
  const mocksdk = new firebasemock.MockFirebaseSdk(
    path => {
      return path ? mockdatabase.child(path) : mockdatabase;
    },
    () => {
      return mockauth;
    }
  );

  const firebase = mocksdk.initializeApp(); // can take a path arg to database url
  // optional - expose the mock
  global.firebase = firebase;

  // return the mock to match your export api
  return firebase;
});

describe(`employer postings component`, () => {
  it(`should render without error`, () => {
    const tree = renderer.create(<EmployerPostings />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
