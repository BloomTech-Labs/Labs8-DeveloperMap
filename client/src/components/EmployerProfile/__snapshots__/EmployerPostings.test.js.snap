// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`employer postings component should render without error 1`] = `
<div
  className="sc-bxivhb lnIAlR"
>
  <div
    className="sc-htpNat daVAfG"
  >
    <h1>
      This company has no job postings.
    </h1>
  </div>
</div>
`;
